synapse/storage/databases/main/event_push_actions.py

```diff
@@ -86,6 +99,27 @@ def __init__(self, database: DatabasePool, db_conn, hs):
         self._rotate_delay = 3
         self._rotate_count = 10000
 
+    def _stream_ordering_from_event_id_and_room_id_txn(
```

Comment:
note that something very similar to this has just been added in #8112. It would be good to avoid adding a second copy.